portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/site/PortalSiteHelperImpl.java

```diff
@@ -350,7 +350,8 @@ private List<Map<String, Object>> getSiteMaps(Collection<Site> sites, String cur
 		// Precompute a mapping from parent site IDs to child site IDs.
 		Map<String, List<Map<String, String>>> parentToChildSites;
 		if (!Arrays.asList("false", "never").contains(serverConfigurationService.getString("portal.includesubsites", "false"))) {
-			parentToChildSites = sites.stream()
+			parentToChildSites = siteService.getSites(org.sakaiproject.site.api.SiteService.SelectionType.ACCESS,
+					null, null, null, org.sakaiproject.site.api.SiteService.SortType.TITLE_ASC, null).stream()
```

Comment:
are you throwing away the sites passed into this method? if yes, you should remove it from the method signature. or change the calling method to set a different `sites`